bookwyrm/static/js/status_cache.js

```diff
@@ -90,6 +90,13 @@ let StatusCache = new class {
             trigger.removeAttribute('disabled');
         })
         .then(response => {
+            if (response.headers.get("forceReload")) {
```

Comment:
I wonder if it's possible to toggle whether the button uses ajax or a regular submit, rather than doing the ajax and then forcing a reload? It checks if the `submit-status` class is present on the form to attach the javascript listener. If that class is absent, it's just a regular old form submission.

Reply:
That would certainly feel less like a hacky workaround!

Reply:
We can't just remove `submit-status` because it's coming from the modals and we don't (I mean, unless we do...) want to force a refresh on the `shelve_button`. I've adjusted this to require less code.